### ISSUE TITLE
More expressive error messsage possible for "Expected string, received string"

### ISSUE THREAD

**Issue by Gregoor** [colinhacks/zod#1045]
Thanks for zod, I love it!

[Example:](https://runkit.com/gregoor/zod-string-literal-error)
```js
const { z } = require("zod");
const Special = z.literal("wat");
console.log(Special.safeParse("nah").error.issues[0].message);
```
which outputs `Expected string, received string`

Wouldn't `Expected "wat", received "nah"` be more preferable? I suppose there is some nuance and edge cases where this isn't feasible?

**Comment by Gregoor** [comment]
Yeah I think a differentiation there for literals would be great. With that pointer of yours, and a comment from a contributor whether this would be merge-worthy I'd be happy to take a stab at a PR.

**Comment by andresgutgon** [comment]
It should be easy, go for it imo : )

**Comment by andresgutgon** [comment]
I think this fix this issue https://github.com/colinhacks/zod/pull/1052

**Comment by FlorianWendelborn** [comment]
@Gregoor can you close this issue? It seems like it was implemented in #1052.

**Comment by Gregoor** [comment]
For me #1052 is not a sufficient fix, I'd like to know the exact value mismatch. I had an idea over in my PR for a fix that would address it for me:

> Makes sense, thanks for the info, I can see how not having this in is a better security default. That said, for cases where it matters less like mine, I was thinking of turning `errorIncludesInputData` once and if #851 lands. Would you be open to a PR that builds on that to show literal values if that flag is set?

That said, ofc I'd leave it to the maintainers to close this issue if that feels sufficiently adressed / not important / out of scope.

**Comment by EdsterG** [comment]
There's currently an open effort to support exactly what you're looking for @Gregoor but it may have slipped through the cracks. https://github.com/colinhacks/zod/pull/851

**Comment by Gregoor** [comment]
Yeah I'm very much cheering for that one, which is why I offered to extend my PR to only expose literal values if that flag, once #851 lands. I have not tried this PR locally yet, but my assumption is that it still would not log the expected literal value, so another patch would be necessary for my issue to be solved.
